codegen/table.go

```diff
@@ -90,5 +92,50 @@ func NewTableFromStruct(name string, obj interface{}, opts ...TableOption) (*Tab
 		}
 	}
 
-	return t, nil
+	// add PK options
+	columns := make(ColumnDefinitions, 0, len(t.Columns))
+	for _, column := range t.Columns {
+		if _, ok := t.extraPKColumns[column.Name]; ok {
+			column.Options.PrimaryKey = true
+		}
+		columns = append(columns, column)
+	}
+	t.Columns = columns
```

Comment:
I think this should raise an error if a specified PK column was not found, since it would indicate a configuration error

Reply:
Nice point